app/goal_routes.py

```diff
@@ -0,0 +1,122 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from .task_routes import validate_model
+import requests
+
+goals_bp = Blueprint("goals",__name__,url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(title=request_body["title"])
+
+    except:
+        abort(make_response({
+        "details": "Invalid data"
```

Comment:
Like we validated models, could we also make a general function that validates request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.

Reply:
Now we can have error handling and request body validation across all our routes more easily.